doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -1023,6 +1023,14 @@ Requesting information
    fixpoint and cofixpoint is violated at some time of the construction
    of the proof without having to wait the completion of the proof.
 
+.. cmd:: Validate Proof
+
+   This command checks that the current partial proof is well-typed.
+   It is useful to track down tactic bugs as without it errors are
+   delayed until the proof is complete.
+
+   It does not check the guard condition, use :cmd:`Guarded` for that.
```

Comment:
```suggestion
   It does not check the guard condition.  Use :cmd:`Guarded` for that.
```

We should define "guard condition" and put it in the glossary.  If you're willing, would be great to add that definition in this PR and contrast it to type checking.  Up to you.